### PR TITLE
enhancement(observability): Convert `EventsReceived` to a registered event

### PR DESCRIPTION
**Pull request by bruceg**
This is the last core event conversion I am planning on doing. There is still some remaining work on `BytesSent` which I will probably take a stab at, but that had ended up being more convoluted than I had hoped, as well as the ARC metrics. The remainder will be smaller scoped, and probably be handled at the integration level.

Epic: #13691

This ended up rewriting a chunk of the `redis` source, which flowed out of `fn handle_line` having too many parameters, and then recognizing the shared parameters being passed around to the different handler types.

### CODE REVIEW

File: src/sources/util/message_decoding.rs [hunk 1]
```diff
@@ -27,49 +29,32 @@ pub fn decode_message<'a>(
...
-                            0,
-                            |size, event: &Event| size + event.estimated_json_encoded_size_of(),
-                            move |byte_size| emit!(EventsReceived { byte_size, count }),
-                        ),
-                )
-            }
+                        }
+                    }
+                }
+                events_received.emit(CountByteSize(1, event.estimated_json_encoded_size_of()));
```
Comment (neuronull) [inline src/sources/util/message_decoding.rs:55]: It looks like `count` used to be taken from the `events.len()` before your changes?
Why is it changing to a hard coded ` here? Is it always going to be one event? If so why do we need the `into_iter()` above?

File: src/sources/amqp.rs [hunk 2]
```diff
@@ -322,10 +326,10 @@ async fn receive_event(
...
                     });
 
-                    emit!(EventsReceived {
-                        byte_size: events.estimated_json_encoded_size_of(),
-                        count: events.len(),
-                    });
+                    events_received.emit(CountByteSize(
+                        events.len(),
+                        events.estimated_json_encoded_size_of(),
+                    ));
```
[the remark this answers is not in the document]
Comment (bruceg) [reply src/sources/amqp.rs:332]: The members, as you surmised, are `(count, byte size)` as described by the name. A comment would be a nice improvement yes.

[the remark this answers is not in the document]
Comment (neuronull) [reply src/sources/amqp.rs:332]: Whoa. I didn't even notice that about the struct name ! I feel unobservant now.

File: src/sources/util/message_decoding.rs (hunk 1 above)
Comment (neuronull) [reply src/sources/util/message_decoding.rs:55]: Ah, I see it now. Thanks for the clarity. Need to use that ignore whitespace feature more often.

**Review by neuronull — APPROVED** [review]
LGTM

**Review by StephenWakely — APPROVED** [review]

**Pull request merged**